src/main/java/taxi/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createCar(resultSet, connection));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find car with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> carList = new ArrayList<>();
+        String getAllQuery = "SELECT c.id AS car_id, model, "
+                + "m.manufacturer_name AS manufacturer_name, "
+                + "m.manufacturer_country AS manufacturer_country, "
+                + "m.id AS manufacturer_id "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(createCar(resultSet, connection));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find any drivers", e);
+        }
+        return carList;
+    }
+
+    private Car createCar(ResultSet resultSet, Connection connection) {
+        try {
+            Long carId = resultSet.getObject("car_id", Long.class);
+            String carModel = resultSet.getObject("model", String.class);
+            String manufacturerName = resultSet.getObject("manufacturer_name", String.class);
+            String manufacturerCountry = resultSet.getObject("manufacturer_country", String.class);
+            Long manufacturersId = resultSet.getObject("manufacturer_id", Long.class);
+            Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+            manufacturer.setId(manufacturersId);
+            Car newCar = new Car(carModel, manufacturer);
+            newCar.setDrivers(getAllDriversByCarId(carId, connection));
+            newCar.setId(carId);
+            return newCar;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a 'car', "
+                    + "can not parse data", e);
+        }
+    }
+
+    private List<Driver> getAllDriversByCarId(Long carId, Connection connection) {
+        String selectQuery = "SELECT d.id, d.driver_name, d.license_number "
+                + "FROM cars_drivers "
+                + "LEFT JOIN drivers d ON d.id = cars_drivers.driver_id "
+                + "LEFT JOIN cars c ON c.id = cars_drivers.car_id "
```

Comment:
do we need to join cars here? and why left join?

Reply:
No we do not need to `join` `cars` here, replaced `left join` just with `join` as we do not need null rows which did not match